did-vc.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Handles DID and VC logic for ID Hub integration.
+ *
+ * @package Civil_Publisher
+ */
+
+namespace Civil_Publisher;
+
+/**
+ * Init DID logic.
+ */
+function init() {
+	add_filter( 'template_include', __NAMESPACE__ . '\include_template' );
+	add_filter( 'init', __NAMESPACE__ . '\rewrite_rules' );
+
+	if ( current_user_can( 'manage_options' ) && empty( get_option( DID_RSA_PRIVATE_KEY ) ) ) {
+		try {
+			init_key_pair();
+		} catch ( Exception $e ) {
+			// @TODO/tobek Surface this error in admin. If missing openssl configs are a problem in the wild, consider using phpseclib instead.
+			update_option( DID_RSA_KEY_ERROR, 'Failed to generate openssl key pair: ' . $e->getMessage() );
+		}
+	}
+}
+
+/**
+ * Init DID key pair and save in database.
+ *
+ * Note that we store the DID private key in plain text in the database. This is of course awfully insecure, but if we want to be able to publish VCs without user interaction then the plugin needs access to the private key somehow. We could do various things to superficially harden the key but they are all either too brittle (e.g. using user's login password as encryption key would break if user loses password and resets it, while using wp-config.php salts as encryption key would break if user migrates to a new WP install) or trivial to defeat (storing encryption key in plugin to encrypt value stored in DB sounds nice, but anyone with access to the DB could trivially google the option name and find this open source plugin and decrypt from there). Even something fancy that required user-held secrets and user interaction for every action (e.g. signing in metamask) could be vulnerable to spoofed requests if the site is compromised. Fundamentally this private key does not protect anything special at the moment; it's more of an identifier. This can be revisited in the future.
+ */
+function init_key_pair() {
+	$key = openssl_pkey_new(
+		array(
+			'digest_alg' => 'sha512',
+			'private_key_bits' => 4096,
+			'private_key_type' => OPENSSL_KEYTYPE_RSA,
```

Comment:
can this use ECDSA instead? I think its ok if we don't use the secp256k1 curve (ethereum) - p-256/sha256 is prob fine. This more common and easier to validate

Reply:
Discussed in slack DM just updating here, non-secp256k1 ECDSA isn't supported by DID spec, so RSA is only option for something widely-supported on PHP (except maybe GPG or JWS), but this whole thing could change depending DAF integration.